zest/releaser/longtest.py

```diff
@@ -1,40 +1,49 @@
 """Do the checks and tasks that have to happen before doing a release.
 """
+try:
+    from docutils.core import publish_string
+    HAVE_DOCUTILS = True
+except ImportError:
+    HAVE_DOCUTILS = False
 import logging
-import os
 import sys
 import tempfile
 import webbrowser
 
 from zest.releaser import utils
-from zest.releaser.utils import execute_command
+from zest.releaser.utils import _execute_command
 
 logger = logging.getLogger(__name__)
 
 
 def show_longdesc():
-    filename1 = tempfile.mktemp()
-    filename2 = tempfile.mktemp()
-    filename2 = filename2 + '.html'
+    if not HAVE_DOCUTILS:
+        # Stupid docutils throws SystemExit
+        logging.error(
+            'Error generating html. Please install docutils (or zc.rst2).')
+        sys.exit(1)
+
+    filename = tempfile.mktemp(u'.html')
     # Note: for the setup.py call we use execute_command() from our
     # utils module. This makes sure the python path is set up right.
     # For the other calls we use os.system(), because that returns an
     # error code which we need.
-    execute_command(utils.setup_py('--long-description > %s' %
-                          filename1))
-    error = os.system('rst2html.py %s > %s' % (filename1, filename2))
-    if error:
-        # On Linux it needs to be 'rst2html', without the '.py'
-        error = os.system('rst2html %s > %s' % (filename1, filename2))
-    if error:
-        # Alternatively, zc.rst2 provides rst2 xyz.
-        error = os.system('rst2 html %s > %s' % (filename1, filename2))
-    if error:
+    longdesc = _execute_command(utils.setup_py([u'--long-description']))
+    try:
+        rst = publish_string(
+            source=longdesc,
+            writer_name='html',
+            enable_exit_status=False)
+    except SystemExit:
+        # Stupid docutils throws SystemExit
```

Comment:
For extra fidelity you could use [readme](https://pypi.python.org/pypi/readme), which is currently used by PyPI (as well as its successor Warehouse) to render long_description strings.

I've encountered long_descriptions that would render fine with plain docutils but fail on PyPI.